applications/vanilla/models/class.commentmodel.php

```diff
@@ -1273,11 +1273,11 @@ public function save($formPostValues, $settings = false) {
                     $this->fireEvent('AfterSaveComment');
                 }
             }
-        }
 
-        // Update discussion's comment count.
-        if (isset($formPostValues['DiscussionID'])) {
-            $this->updateCommentCount($formPostValues['DiscussionID'], ['Slave' => false]);
+            // Update discussion's comment count.
+            if (isset($formPostValues['DiscussionID'])) {
+                $this->updateCommentCount($formPostValues['DiscussionID'], ['Slave' => false]);
+            }
         }
```

Comment:
Was this change intended to make it into this PR?

Reply:
It was sort of related. The code before was updating the count regardless of whether or not the comment validated. There were some issues with invalid comments calling this code and making some odd errors. However, I'm guessing after further changes in my PR that no longer happened. Regardless, you can see the if statement I'm putting this in seems very sensible.